doc/reference/usermode/syscalls.rst

```diff
@@ -197,17 +170,19 @@ necessary privilege elevation. In this layer, all arguments are treated as an
 unsigned 32-bit type. There is always a 32-bit unsigned return value, which
 may or may not be used.
 
-Some system calls may have more than six arguments. The number of arguments
-passed via registers is fixed at six for all architectures. Additional
-arguments will need to be passed in a struct, which needs to be treated as
-untrusted memory in the handler function. This is done by the derived
-functions :c:func:`_syscall_invoke7` through :c:func:`_syscall_invoke10`.
+Some system calls may have more than six arguments. The number of
+arguments passed via registers is fixed at six for all
+architectures. Additional arguments will need to be passed in an array
+in the source memory space, which needs to be treated as untrusted
+memory in the handler function. This code (packing, unpacking and
+validation) is generated automatically as needed in the stub above and
+in the unmarshalling function.
 
-Some system calls may return a value that will not fit in a 32-bit register,
-such as APIs that return a 64-bit value. In this scenario, the return value is
-populated in a memory buffer that is passed in as an argument. For example,
-see the implementation of :c:func:`_syscall_ret64_invoke0` and
-:c:func:`_syscall_ret64_invoke1`.
+Some system calls may return a value that will not fit in a 32-bit
+register, such as APIs that return a 64-bit value. In this scenario,
+the return value is populated in a (untrusted!) memory buffer that is
```

Comment:
how about, ``populated in an **untrusted** memory buffer`` (use bold for emphasis instead of the parentheses and exclamation point.